src/librustc/middle/resolve_lifetime.rs

```diff
@@ -860,7 +860,9 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
         });
     }
 
-    fn resolve_lifetime_ref(&mut self, lifetime_ref: &hir::Lifetime) {
+    fn resolve_lifetime_ref(&mut self,
+                            lifetime_ref: &hir::Lifetime,
+                            context_ref: Option<&hir::Lifetime>) {
```

Comment:
The terminology here is wrong IMO. The intention of `_ref`, AFAICT, is to mean "lifetime reference", as in, one use of  (referring to) a lifetime. So this extra argument should be `maybe_def` or some such, but even then, I would just bypass this whole function from `check_lifetime_defs`, for `'static`, i.e. handle it there instead of here.